### PR TITLE
Few coverage nitpicks for the cmath module

### PR DESCRIPTION
**Pull request by skirpichev**
- partial tests for cosh/sinh overflows (L535 and L771).  I doubt both ||-ed conditions could be tested.
- removed inaccessible case in sqrt (L832): ax=ay=0 is handled above (L823) because fabs() is exact.  Also added test (checked with mpmath and gmpy2) for second condition on that line.
- some trivial tests for isclose (cover all conditions on L1217-1218)
- add comment for uncovered L1018

### CODE REVIEW

**Comment by Turesinndaniss** [comment]
> * partial tests for cosh/sinh overflows (L535 and L771).  I doubt both ||-ed conditions could be tested.
> * removed inaccessible case in sqrt (L832): ax=ay=0 is handled above (L823) because fabs() is exact.  Also added test (checked with mpmath and gmpy2) for second condition on that line.
> * some trivial tests for isclose (cover all conditions on L1217-1218)
> * add comment for uncovered L1018

**Comment by skirpichev** [comment]
BTW, there are some places with workarounds for systems with unsigned zeros (e.g. [this](https://github.com/python/cpython/blob/6f25657b83d7a680a97849490f6e973b3a695e1a/Modules/cmathmodule.c#L237-L242) or [this](https://github.com/python/cpython/blob/6f25657b83d7a680a97849490f6e973b3a695e1a/Modules/cmathmodule.c#L440)).  Does it make sense to keep that after #102046?

File: Lib/test/cmath_testcases.txt [hunk 1]
```diff
@@ -1536,6 +1536,7 @@ sqrt0141 sqrt -1.797e+308 -9.9999999999999999e+306 -> 3.7284476432057307e+152 -1
 sqrt0150 sqrt 1.7976931348623157e+308 0.0 -> 1.3407807929942596355e+154 0.0
 sqrt0151 sqrt 2.2250738585072014e-308 0.0 -> 1.4916681462400413487e-154 0.0
 sqrt0152 sqrt 5e-324 0.0 -> 2.2227587494850774834e-162 0.0
+sqrt0153 sqrt 5e-324 1.0 -> 0.7071067811865476 0.7071067811865475
```
Comment (mdickinson) [inline Lib/test/cmath_testcases.txt]: Fix to make the imaginary part what we'd get if we had perfectly correct rounding:

```suggestion
sqrt0153 sqrt 5e-324 1.0 -> 0.7071067811865476 0.7071067811865476
```

**Review by mdickinson — APPROVED** [review]
LGTM; I took the liberty of fixing the sqrt test value. (The values in `cmath_testcases.txt` should all be the "ideal" values - i.e., those that would arise from a perfectly correctly-rounded implementation.)

**Comment by mdickinson** [comment]
> Does it make sense to keep that after #102046?

Let's leave as is - they're not causing any problems, so there's no reason to change. And who knows - we _might_ encounter some situation in the future where we want this code to work on systems without signed zeros ...

**Comment by skirpichev** [comment]
> Let's leave as is - they're not causing any problems, so there's no reason to change.

Then it probably does make sense
1. to document branch cuts conventions for non-IEEE 754 systems (in sources, not user docs)
2. and/or add comments in other places with similar workarounds.  I doubt this stuff will be obvious for a reader, coming with the current module docs in mind.

**Comment by mdickinson** [comment]
> Then it probably does make sense [...]

No, sorry, not worth it. We can worry about that if a relevant system ever shows up.

**Pull request merged**